src/Operation/Distinct.php

```diff
@@ -22,47 +22,60 @@
 final class Distinct extends AbstractOperation
 {
     /**
-     * @return Closure(callable(mixed): (Closure(mixed): bool)): Closure(callable(T, TKey): mixed): Closure(Iterator<TKey, T>): Generator<TKey, T>
+     * @template U
+     *
+     * @return Closure(callable(U): Closure(U): bool): Closure(callable(T, TKey): U): Closure(Iterator<TKey, T>): Generator<TKey, T>
      */
     public function __invoke(): Closure
     {
         return
             /**
-             * @param callable(mixed): (Closure(mixed): bool) $comparatorCallback
+             * @param callable(U): (Closure(U): bool) $comparatorCallback
              *
-             * @return Closure(callable(T, TKey): mixed): Closure(Iterator<TKey, T>): Generator<TKey, T>
+             * @return Closure(callable(T, TKey): U): Closure(Iterator<TKey, T>): Generator<TKey, T>
              */
             static fn (callable $comparatorCallback): Closure =>
                 /**
-                 * @param callable(T, TKey): mixed $accessorCallback
+                 * @param callable(T, TKey): U $accessorCallback
                  *
                  * @return Closure(Iterator<TKey, T>): Generator<TKey, T>
                  */
                 static function (callable $accessorCallback) use ($comparatorCallback): Closure {
+                    /**
+                     * @param callable(T, TKey): U $accessorCallback
+                     *
+                     * @return Closure(callable(U): Closure(U): bool): Closure(list<array{0: TKey, 1: T}>, array{0: TKey, 1: T}): list<array{0: TKey, 1: T}>
+                     */
```

Comment:
this as well as the return type [here](https://github.com/loophp/collection/pull/111/commits/c0fca86fb35e00fac05d8e9288b8f6b301e1bafc#diff-3b63570fb11bc48cd14c56240059ece0e699ed0f466f9ea7ab7c10ba9c3616abR54) were the missing pieces that were causing the Psalm static analysis errors

Reply:
I got it now, my bad!